src/algorithms/KirlikSayin.jl

```diff
@@ -0,0 +1,241 @@
+
+## Supported optimizer attributes
+
+"""
+
+mutable struct KirlikSayin <: AbstractAlgorithm
+  #yI::Union{Nothing, Vector{Float64}}
+  #yN::Union{Nothing, Vector{Float64}}
+  #L::Vector{Rectangle}
+
+  #KirlikSayin() = new(nothing, nothing, Rectangle[])
+  KirlikSayin() = new()
+end
+
+#=
+MOI.supports(::KirlikSayin, ::IdealPoint) = true
+
+function MOI.get(alg::KirlikSayin, attr::IdealPoint)
+  return alg.yI
+end
+
+function MOI.set(alg::KirlikSayin, attr::IdealPoint, value)
+  alg.yI = value
+  return
+end
+
+function MOI.set(alg::KirlikSayin, attr::IdealPoint, index, value)
+  alg.yI[index] = value
+  return
+end
+
+MOI.supports(::KirlikSayin, ::NadirPoint) = true
+
+function MOI.get(alg::KirlikSayin, attr::NadirPoint)
+  return alg.yN
+end
+
+function MOI.set(alg::KirlikSayin, attr::NadirPoint, value)
+  alg.yN = value
+  return
+end
+
+function MOI.set(alg::KirlikSayin, attr::NadirPoint, index, value)
+  alg.yN[index] = value
+  return
+end
+=#
+struct Rectangle
+  l::Vector{Float64}
+  u::Vector{Float64}
+  dim::Int
+  function Rectangle(l::Vector{Float64}, u::Vector{Float64})
+    @assert length(l) == length(u) "Dimension mismatch between l and u"
+    return new(l, u, length(l))
+  end
+end
+
+function ⊆(Rᵢ::Rectangle, Rⱼ::Rectangle)
+	@assert Rᵢ.dim == Rⱼ.dim "Dimension mismatch"
+	all(Rᵢ.l .≥ Rⱼ.l) && all(Rᵢ.u .≤ Rⱼ.u)
+end
+
+function remove_rect!(L::Vector{Rectangle}, R::Rectangle)
+  ix_to_pop = []
+  for (t, Rₜ) in enumerate(L)
+    if Rₜ ⊆ R
+      push!(ix_to_pop, t)
+    end
+  end
+  # Lᵤ = L[ix_to_pop]
+  deleteat!(L, ix_to_pop)
+  # return Lᵤ
+end
+
+function split_rectangle(r::Rectangle, axis::Int, f::Float64)
+  l = [i != axis ? r.l[i] : f for i = 1:r.dim]
+  u = [i != axis ? r.u[i] : f for i = 1:r.dim]
+  return Rectangle(r.l, u), Rectangle(l, r.u)
+end
+
+function update_list(L::Vector{Rectangle}, f::Vector{Float64})
+  @info L
+  @info f
+  L̄, L = L, Vector{Rectangle}()
+  for Rᵢ in L̄
+    lᵢ, uᵢ = Rᵢ.l, Rᵢ.u
+    T = [Rᵢ]
+    for j = 1:length(f)
+      if lᵢ[j] < f[j] < uᵢ[j]
+        @info "anything?"
+        T̄ = Vector{Rectangle}()
+        for Rₜ in T
+          push!(T̄, split_rectangle(Rₜ, j, f[j])...)
+        end
+        T = T̄
+      end
+    end
+    push!(L, T...)
+  end
+  return L
+end
+
+#=
+=#
+function optimize_multiobjective!(algorithm::KirlikSayin, model::Optimizer)
+  
+  # Problem with p objectives.
+  # Set k = 1, meaning the nondominated points will get projected 
+  # down to the objective {2, 3, ..., p}
+
+  k = 1
+
+  XE, YN = Vector{Dict{MOI.VariableIndex,Float64}}(), Vector{Vector{Float64}}()
+
+  solutions = ParetoSolution[]
+  
+  variables = MOI.get(model.inner, MOI.ListOfVariableIndices())
+  sense = MOI.get(model.inner, MOI.ObjectiveSense())
+  nbObj = MOI.output_dimension(model.f)
+  yI, yN = zeros(nbObj), zeros(nbObj)
+  δ = ifelse(sense == MOI.MIN_SENSE, -1, 1)
+
+  # Ideal and Nadir point estimation
+  for (i, f_i) in enumerate(MOI.Utilities.scalarize(model.f))
```

Comment:
I've just noticed we have `MOI.Utilities.scalarize` and `MOI.Utilities.eachscalar`... (cc @blegat)

I don't have a preference, but you call `scalarize` a few times below, so it's probably best if we call it once and cache the result.